File: src/ca/mindmagic/game/map/grid/shape/Triangle.java
package ca.mindmagic.game.map.grid.shape;

import java.awt.Point;

public class Triangle extends Shape {

  public Triangle(){
    super();
  }

  public Triangle(int sideLength){
    super(sideLength);
  }

  @Override public int numberOfSides() {
    return 3;
  }

  @Override public Point[] getVertices() {
    return getVertices(sideLength);
  }

  public static Point[] getVertices(int sidelength){
    Point[] vertices = new Point[3];
    vertices[0] = new Point();  // TODO create the appropriate points
    vertices[1] = new Point();
    vertices[2] = new Point();
    return vertices;
  }
}

File: src/ca/mindmagic/game/map/grid/pattern/Triangle.kt
package ca.mindmagic.game.map.grid.pattern

import javafx.geometry.Point2D

class Triangle(sidelength: Double) : Shape(sidelength) {



    override fun numberOfSides(): Int {
        return 3
    }

    override fun getVertices(): Array<Point2D> {
        return getVertices(sideLength)
    }

    companion object {

        fun getVertices(sidelength: Double): Array<Point2D> {
            return arrayOf(
                    Point2D(0.0, 0.0),
                    Point2D(0.0, 0.0),
                    Point2D(0.0, 0.0)
            )
        }
    }
}
